src/Adapters/Files/S3Adapter.js

```diff
@@ -0,0 +1,76 @@
+// S3Adapter
+//
+// Stores Parse files in AWS S3.
+
+import * as AWS from 'aws-sdk';
+import { FilesAdapter } from './FilesAdapter';
+
+const DEFAULT_S3_REGION = "us-east-1";
+const DEFAULT_S3_BUCKET = "parse-files";
+
+export class S3Adapter extends FilesAdapter {
+  // Creates an S3 session.
+  // Providing AWS access and secret keys is mandatory
+  // Region and bucket will use sane defaults if omitted
+  constructor(accessKey, secretKey, options = {}) {
+    super();
+
+    this._region = options.region || DEFAULT_S3_REGION;
+    this._bucket = options.bucket || DEFAULT_S3_BUCKET;
+    this._bucketPrefix = options.bucketPrefix || '';
+    this._directAccess = options.directAccess || false;
+
+    let s3Options = {
+      accessKeyId: accessKey,
+      secretAccessKey: secretKey,
+      params: { Bucket: this._bucket }
+    };
+    AWS.config._region = this._region;
+    this._s3Client = new AWS.S3(s3Options);
+  }
+
+  // For a given config object, filename, and data, store a file in S3
+  // Returns a promise containing the S3 object creation response
+  createFileAsync(config, filename, data) {
```

Comment:
In JS Async is usually assumed to be the default, so you have `doSomething` and `doSomethingSync` instead of `doSomethingAsync` and `doSomething`. I don't think this interface is public yet so it may not be too late to change it. Up to you.